### PR TITLE
Re-add linkopts and set -Wl,-headerpad_max_install_names on macos

### PR DESCRIPTION
**Pull request by jvolkman**
When building artifacts for macos, pass the `-Wl,-headerpad_max_install_names` linker option to leave space for modifying things like `rpath`.

The built-in toolchain also does it [here](https://github.com/bazelbuild/bazel/blob/0a10409e90f4f6301c24b3743cce90ad8012a81f/tools/cpp/unix_cc_configure.bzl#L599).

Also fixed an issue where the `.map` files under `glibc-hacks` weren't being included in the list of toolchain files.

### CODE REVIEW

**Comment by motiejus** [comment]
Thank you. Let's wait until #30 is merged (probably EOW or beginning of next week) and then rebase. I will then properly review it.

I am OK with  `-Wl,-headerpad_max_install_names` though.

**Comment by motiejus** [comment]
> If glibc-hacks is going away, would you like me to just remove that bit from this PR?

That works too.

**Comment by jvolkman** [comment]
Alright, I dropped that second commit. It should just be the headerpad change now.

**Comment by motiejus** [comment]
Sorry for the delay and all the stuff that piled up. Would you mind rebasing?

**Comment by jvolkman** [comment]
> Sorry for the delay and all the stuff that piled up. Would you mind rebasing?

Done!

**Review by motiejus — APPROVED** [review]

**Pull request merged**
